### PR TITLE
fix(components/toast): toast shows icon when default state is being used

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

File: libs/components/toast/src/lib/modules/toast/fixtures/toast-with-toaster-service.component.fixture.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { SkyToasterService } from '../toaster.service';
 
 @Component({
   selector: 'sky-test-cmp',
-  templateUrl: './toast.component.fixture.html',
+  templateUrl: './toast-with-toaster-service.component.fixture.html',
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/toast/src/lib/modules/toast/fixtures/toast-with-toaster-service.component.fixture.ts:11]: I found this when working... we actually had the `html` file but the fixture was pointing at the wrong one. I didn't research the existing tests any more but they still passed when I changed it. We may just not need this fixture (shrug)

File: libs/components/toast/src/lib/modules/toast/toast.component.ts [hunk 2]
```diff
@@ -45,13 +46,18 @@ export class SkyToastComponent implements OnInit, OnDestroy {
    * Specifies a `SkyToastType` type for the toast to determine the color and icon to display.
    */
   @Input()
-  public set toastType(value: SkyToastType) {
-    this._toastType = value;
+  public set toastType(value: SkyToastType | undefined) {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/toast/src/lib/modules/toast/toast.component.ts:49]: I did a couple strict mode fixes in this file such as getting rid of the getter here and using the `OrDefault` convention. However, I did not strict mode everything including doing private private things as that was not in scope and there are a lot of violations in this component. I started to do them and it became too much for this unrelated fix and so I backed them out.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/426/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/426/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/426/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/426/playground/)

File: libs/components/toast/src/lib/modules/toast/toast.component.ts [hunk 3]
```diff
@@ -45,13 +46,18 @@ export class SkyToastComponent implements OnInit, OnDestroy {
...
+    if (value === undefined) {
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/toast/src/lib/modules/toast/toast.component.ts]: Did strict mode prevent you from using `this.toastTypeOrDefault = value || SKY_TOAST_TYPE_DEFAULT`? If so, I'd prefer the ternary operator here:

```
this.toastTypeOrDefault = value === undefined ? SKY_TOAST_TYPE_DEFAULT : value;
```

File: libs/components/toast/src/lib/modules/toast/toast.component.ts [hunk 4]
```diff
@@ -45,13 +46,18 @@ export class SkyToastComponent implements OnInit, OnDestroy {
...
+      this.toastTypeOrDefault = SKY_TOAST_TYPE_DEFAULT;
+    } else {
+      this.toastTypeOrDefault = value;
+    }
     this.updateIcon();
-  }
 
-  public get toastType(): SkyToastType {
-    return this._toastType === undefined ? SkyToastType.Info : this._toastType;
+    this.ariaLive =
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/toast/src/lib/modules/toast/toast.component.ts]: Wouldn't the ARIA live and ARIA role properties also need to be set when the initial state of the component is established with the default toast type? If so, I'd rename `updateIcon()` to `updateForToastType()` and move this logic there.

File: libs/components/toast/src/lib/modules/toast/toast.component.ts [hunk 5]
```diff
@@ -112,12 +116,12 @@ export class SkyToastComponent implements OnInit, OnDestroy {
...
   private ngUnsubscribe = new Subject<void>();
 
-  private _toastType: SkyToastType;
-
   constructor(
     private changeDetector: ChangeDetectorRef,
     @Optional() private toasterService?: SkyToasterService
-  ) {}
+  ) {
+    this.updateIcon();
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/toast/src/lib/modules/toast/toast.component.ts]: I'd prefer to see this moved to `ngOnInit()` with the rest of the logic that initializes the state of the component, such as the `isOpen` state.

File: libs/components/toast/src/lib/modules/toast/toast.component.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/toast/src/lib/modules/toast/toast.component.ts]: Nope. That was just me coding too fast.

File: libs/components/toast/src/lib/modules/toast/toast.component.ts (hunk 5 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/toast/src/lib/modules/toast/toast.component.ts]: Done

File: libs/components/toast/src/lib/modules/toast/toast.component.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/toast/src/lib/modules/toast/toast.component.ts]: Fixed

File: libs/components/toast/src/lib/modules/toast/toast.component.ts (hunk 4 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/toast/src/lib/modules/toast/toast.component.ts]: In theory - the defaults that I set on the fields are all that are needed as they are correct for the default. However, I don't think what you suggested hurts and does give us some confidence so I went ahead and did that.

**Review by Blackbaud-PaulCrowder — APPROVED** [review]
